projects/charger/src/generic_can_hw.c

```diff
@@ -0,0 +1,119 @@
+#include "generic_can_hw.h"
+
+#include <stddef.h>
+#include <stdint.h>
+#include <string.h>
+
+#include "can_hw.h"
+#include "event_queue.h"
+#include "generic_can.h"
+#include "generic_can_helpers.h"
+#include "generic_can_msg.h"
+#include "log.h"
+#include "soft_timer.h"
+#include "status.h"
+
+#define CAN_BUS_OFF_RECOVERY_TIME_MS 500
+
+static GenericCanInterface s_interface;
+
+// CANHwEventHandlerCb: Tx Occurred (dummy to avoid segfault if not present).
+static void prv_tx_handler(void *context) {
```

Comment:
It segfaults?

Reply:
Yeah, if I don't have a handler for TX ready it will segfault.

Reply:
Weird, we should have NULL checks.